homeassistant/components/flux_led/light.py

```diff
@@ -235,6 +235,11 @@ def __init__(self, unique_id: str, device: dict, effect_speed: int):
         self._get_rgbw = None
         self._get_rgb = None
 
+        try:
+            self._bulb = WifiLedBulb(device[CONF_HOST])
+        except BrokenPipeError as error:
```

Comment:
This means that this is doing I/O. Then we should create the bulb outside the entity in the executor and then pass the bulb instance to the entity.

Reply:
Sounds like a good idea... is there an integration I can look at that would help me understand how to do that?

Reply:
I think this should work. Otherwise we can use partial.
```py
bulb = await hass.async_add_executor_job(WifiLedBulb, host)
```

Reply:
OK - does this eliminate the need for a try: to catch errors or does it change the error? (I'm assuming not, just wraps it, just want to be sure).

Reply:
Actually since this is in the `def __init__` it isn't async anyway... I can't await this in the init, is it necessary?

Reply:
We have to do it outside the entity and then pass the bulb to the entity when creating that.

Reply:
And yes, keep the try... except.

Reply:
Gotcha - think I got it.